go-controller/pkg/clustermanager/pod/allocator.go

```diff
@@ -170,10 +182,28 @@ func (a *PodAllocator) releasePodOnNAD(pod *corev1.Pod, nad string, podDeleted,
 
 	uid := string(pod.UID)
 
-	// do not release IPs from the allocator if not flaged to do so or if they
-	// werea already previosuly released
-	if releaseIPsFromAllocator && !a.isPodReleased(nad, uid) {
-		err := a.allocator.ReleaseIPs(a.netInfo.GetNetworkName(), podAnnotation.IPs)
+	hasIPAM := util.DoesNetworkRequireIPAM(a.netInfo)
+	hasIDAllocation := util.DoesNetworkRequireIDs(a.netInfo)
+
+	if !hasIPAM && !hasIDAllocation {
+		// nothing to do
```

Comment:
nit: redundant. I learn nothing from this comment.

Reply:
Reworded